### PR TITLE
Add ValuError message for Adamax

### PR DESCRIPTION
**Pull request by Stealeristaken**
Added valueerror messages for adamax optimizer.

### CODE REVIEW

**Review by awni — COMMENTED** [review]
Thanks! Did you run the pre-commit hooks for formatting? Could you run those and then we can merge?

See the CONTRIBUTING.md doc.

**Comment by Stealeristaken** [comment]
> Thanks! Did you run the pre-commit hooks for formatting? Could you run those and then we can merge?
> 
> See the CONTRIBUTING.md doc.

Done testing on a notebook at local. Changed code positions and deleted betas valueerror messages. It works on local. You can merge

File: python/mlx/optimizers.py [hunk 1]
```diff
@@ -423,6 +423,10 @@ def __init__(
         self, learning_rate: float, betas: List[float] = [0.9, 0.999], eps: float = 1e-8
     ):
         super().__init__(learning_rate, betas, eps)
+        if not 0.0 <= learning_rate:
+            raise ValueError(f"Learning rate should be >=0, {self.learning_rate} was provided instead")
```
Comment (awni) [inline python/mlx/optimizers.py:427]: Is it necessary that the learning rate be `>0`. It seems like a convenient way to do maximization would be to allow positive learning rates. Don't think we should throw in that case

Comment (Stealeristaken) [reply python/mlx/optimizers.py:427]: I have not come across any studies regarding the use of negative learning rates before. When I look at modern frameworks like TensorFlow and PyTorch, they also assume positive learning rates. Of course, as you suggested, we can do it differently, but I believe we can stick with positive learning rates in MLX as other frameworks do. If there are any studies related to this topic, could you please provide the names? I will make the change accordingly

Comment (awni) [reply python/mlx/optimizers.py]: Yea might it might be safer indeed for this to only use positive learning rates. For SGD it's pretty basic:

`p + lr * g` vs `p - lr * g` does maximization vs minimization.

**Review by awni — APPROVED** [review]

**Comment by awni** [comment]
Thanks!! I think it's good the way it is. I don't feel too strongly about learning rate check either way, but thanks for humoring me 😄

**Comment by Stealeristaken** [comment]
> Thanks!! I think it's good the way it is. I don't feel too strongly about learning rate check either way, but thanks for humoring me 😄 

I thank you for the information you provided. I'm relatively new to this, so correcting any mistakes or gaps in my knowledge is beneficial for me. Regards 😁
